src/node/history.h

```diff
@@ -231,8 +231,15 @@ namespace ccf
 
     bool verify(HistoryTree* tree) const
     {
-      return tree->max_index() == path->max_index() && tree->root() == root &&
-        path->verify(root);
+      if (path->max_index() > tree->max_index())
```

Comment:
We try to use {} everywhere, although we haven't found a good way to enforce it so far.

Reply:
Not sure what you're referring to. Do you mean you'd like it to work for empty or all-zero `this->root`?

Reply:
Er no, sorry, what Eddy said about if/else blocks :)